src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,277 @@
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.bwa.BWANativeLibrary;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.CigarUtils;
+import scala.Tuple2;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.RunSGAViaProcessBuilderOnSpark.ContigsCollection;
+import static org.broadinstitute.hellbender.tools.spark.sv.RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID;
+import static org.broadinstitute.hellbender.tools.spark.sv.RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigSequence;
+
+public class ContigAligner implements Closeable {
+
+    static String referencePath;
+
+    final BwaIndex index;
+    final BwaMem bwaMem;
+
+    private static final Logger log = LogManager.getLogger(ContigAligner.class);
+
+    static {
+        BWANativeLibrary.load();
+    }
+
+    public ContigAligner(final String referenceFilePath) throws IOException {
+        referencePath = referenceFilePath;
+
+        try {
+            index = new BwaIndex(LocalizedReference.INSTANCE);
+            log.info("Created BWA index");
+        } catch (final IOException e) {
+            throw new GATKException("Unable to load reference", e);
+        }
+        bwaMem = new BwaMem(index);
+        log.info("Created BWA MEM");
+    }
+
+    public List<AssembledBreakpoint> alignContigs (final ContigsCollection contigsCollection) {
+        final List<AssembledBreakpoint> assembledBreakpoints = new ArrayList<>();
+        try {
+            for(final Tuple2<ContigID, ContigSequence> contigInfo : contigsCollection.getContents()) {
+                final String contigId = contigInfo._1.toString();
+                final byte[] sequence = contigInfo._2.toString().getBytes();
+                final AlnRgn[] alnRgns = bwaAlignSequence(bwaMem, contigId, sequence);
+
+                log.info("alnRgns : " + (alnRgns == null ? "null" : alnRgns.length));
+                // todo: parse cigar for internal indels
+                if (alnRgns.length > 1) {
+
+                    // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
+                    final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
+                            .filter(a -> a.getSecondary() < 0)
+                            .map(AlignmentRegion::new)
+                            .sorted(Comparator.comparing(a -> a.startInAssembledContig))
+                            .collect(arrayListCollector(alnRgns.length));
+
+
+                    final Iterator<AlignmentRegion> iterator = alignmentRegionList.iterator();
+                    if ( iterator.hasNext() ) {
+                        AlignmentRegion current = iterator.next();
```

Comment:
Personally, I find the name "clash" of the two iterators a bit confusing to read.